src/Editor.js

```diff
@@ -536,8 +537,32 @@ const Editor = () => {
                   true
                 );
               }}
+              onDoubleClick={() => setIdOfLayerBeingEdited(id)}
+              onBlur={() => setIdOfLayerBeingEdited(null)}
             >
-              {name}
+              {idOfLayerBeingEdited === id ? (
+                <input
+                  type="text"
+                  autoFocus
+                  onChange={event => {
+                    const updatedName = event.currentTarget.value;
+                    setState(current => ({
+                      ...current,
+                      doc: {
+                        ...current.doc,
+                        layers: current.doc.layers.map(layer => {
+                          return layer.id === id
+                            ? { ...layer, name: updatedName }
+                            : layer;
+                        })
+                      }
+                    }));
+                  }}
+                  value={name}
+                />
+              ) : (
+                name
```

Comment:
I think we need a fallback here incase `name.trim() === ""`. Maybe we just show something like `<unnamed layer>`?